src/pages/index.js

```diff
@@ -68,77 +120,136 @@ const editModalPopup = new PopupWithForm(
   "#profile-edit-modal",
   editProfileModal
 );
+
 addCardPopup.setEventListeners();
 editModalPopup.setEventListeners();
 
+// Event listeners for buttons
 profileAddButton.addEventListener("click", () => {
   addCardPopup.open();
 });
 
-addCardClose.addEventListener("click", () => {
-  addCardPopup.close();
-});
-
 profileEditButton.addEventListener("click", () => {
   const { name, job } = userInfo.getUserInfo();
   editModalTitleInput.value = name;
   editModalSubtitleInput.value = job;
   editModalPopup.open();
 });
 
+// Creating UserInfo instance
 const userInfo = new UserInfo({
   nameSelector: ".profile__title",
   jobSelector: ".profile__subtitle",
+  avatarSelector: ".profile__image",
 });
 
-editModalCloseButton.addEventListener("click", () => {
-  editModalPopup.close();
-});
-
+// Function to handle edit profile form submission
 function editProfileModal() {
   const newName = editModalTitleInput.value;
   const newJob = editModalSubtitleInput.value;
-  userInfo.setUserInfo({ name: newName, job: newJob });
-  editModalPopup.close();
-}
 
+  editModalPopup.renderLoading(true);
+
+  api
+    .updateProfile()
+    .then(() => {
+      console.log("Profile updated successfully:");
+      userInfo.setUserInfo({ name: newName, job: newJob });
+    })
+    .catch((error) => {
+      console.error("Error updating profile:", error);
+    })
+    .finally(() => {
+      editModalPopup.renderLoading(false);
+      editModalPopup.close();
```

Comment:
Needs correction:  Сlose modal in `then` block, otherwise the form will close even if an error occurred and you will need to reopen the form and fill the values again.